app/mixins/paginate.js

```diff
@@ -181,14 +195,182 @@ const PaginateMixin = Mixin.create({
   }
 });
 
-export default PaginateMixin;
+// Class basaed mixin support for glimmer components
+import {
+  action
+} from '@ember/object';
+import {
+  tracked
+} from '@glimmer/tracking';
+import {
+  gt,
+  equal,
+  alias
+} from '@ember/object/computed';
+
+export const PaginationService = superclass => class extends superclass {
```

Comment:
This should ideal be mixin itself...service has a different meaning in Ember.(mainly being singleton)